app/src/main/java/com/github/weg_li_android/MainActivity.kt

```diff
@@ -1,21 +1,55 @@
 package com.github.weg_li_android
 
+import android.Manifest
+import android.app.Activity
+import android.content.ContentResolver
+import android.content.ContentValues
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.net.Uri
+import android.os.Build
 import android.os.Bundle
+import android.os.Environment
+import android.provider.MediaStore
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.DisplayMetrics
+import android.view.Menu
+import android.view.MenuItem
+import android.view.MotionEvent
 import android.view.View
+import android.view.ViewTreeObserver.OnGlobalLayoutListener
 import android.widget.AdapterView
 import android.widget.ArrayAdapter
 import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.view.ActionMode
 import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.app.ActivityCompat
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProviders
+import androidx.recyclerview.widget.RecyclerView
 import com.github.weg_li_android.data.api.ApiHelper
 import com.github.weg_li_android.data.api.ApiServiceImpl
+import com.github.weg_li_android.data.model.Report
 import com.github.weg_li_android.ui.base.ViewModelFactory
 import com.github.weg_li_android.ui.main.viewmodel.MainViewModel
+import com.github.weg_li_android.utils.AutoFitGridLayoutManager
 import kotlinx.android.synthetic.main.activity_main.*
+import timber.log.Timber
+import java.io.OutputStream
+import java.util.*
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), PhotoRecyclerViewAdapter.ItemClickListener, ActivityCompat.OnRequestPermissionsResultCallback {
 
+    private lateinit var photoAdapter: PhotoRecyclerViewAdapter
     private lateinit var mainViewModel: MainViewModel
+    private val pickImage = 1
+    private val takeImage = 2
+    private var mActionMode: ActionMode? = null
```

Comment:
Please let's not use the Hungarian notation, Android Studio tells us the type.  Please change "mActionMode" to "actionMode"